pkg/search/proxy/controller_test.go

```diff
@@ -242,6 +324,416 @@ func TestController_reconcile(t *testing.T) {
 	}
 }
 
+func TestController_Connect(t *testing.T) {
+	var karmadaProxying, clusterProxying, cacheProxying bool
+	ctl := &Controller{
+		karmadaProxy: connectFunc(func(context.Context, schema.GroupVersionResource, string, rest.Responder) (http.Handler, error) {
+			return http.HandlerFunc(func(http.ResponseWriter, *http.Request) {
+				karmadaProxying = true
+			}), nil
+		}),
+		cacheProxy: connectFunc(func(context.Context, schema.GroupVersionResource, string, rest.Responder) (http.Handler, error) {
+			return http.HandlerFunc(func(http.ResponseWriter, *http.Request) {
+				cacheProxying = true
+			}), nil
+		}),
+		clusterProxy: connectFunc(func(context.Context, schema.GroupVersionResource, string, rest.Responder) (http.Handler, error) {
+			return http.HandlerFunc(func(http.ResponseWriter, *http.Request) {
+				clusterProxying = true
+			}), nil
+		}),
+		store: &cacheFuncs{
+			HasResourceFunc: func(gvr schema.GroupVersionResource) bool { return gvr == podGVR },
+		},
+	}
+
+	type args struct {
+		path string
+	}
+	type want struct {
+		karmadaProxying, clusterProxying, cacheProxying bool
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "get api from karmada",
+			args: args{
+				path: "/api",
+			},
+			want: want{
+				karmadaProxying: true,
+			},
+		},
+		{
+			name: "get event api karmada",
+			args: args{
+				path: "/apis/events.k8s.io/v1",
+			},
+			want: want{
+				karmadaProxying: true,
+			},
+		},
+		{
+			name: "list nodes from karmada",
+			args: args{
+				path: "/api/v1/nodes",
+			},
+			want: want{
+				karmadaProxying: true,
+			},
+		},
+		{
+			name: "get node from karmada",
+			args: args{
+				path: "/api/v1/nodes",
+			},
+			want: want{
+				karmadaProxying: true,
+			},
+		},
+		{
+			name: "list pod from cache",
+			args: args{
+				path: "/api/v1/pods",
+			},
+			want: want{
+				cacheProxying: true,
+			},
+		},
+		{
+			name: "list pod from cache with namespace",
+			args: args{
+				path: "/api/v1/namespaces/default/pods",
+			},
+			want: want{
+				cacheProxying: true,
+			},
+		},
+		{
+			name: "get pod from cache",
+			args: args{
+				path: "/api/v1/namespaces/default/pods/foo",
+			},
+			want: want{
+				cacheProxying: true,
+			},
+		},
+		{
+			name: "get pod log from cluster",
+			args: args{
+				path: "/api/v1/namespaces/default/pods/foo/log",
+			},
+			want: want{
+				clusterProxying: true,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			karmadaProxying, clusterProxying, cacheProxying = false, false, false
```

Comment:
Why karmadaProxying, clusterProxying, cacheProxying is always false? A little confused.

Reply:
These variables is to record whether these functions are called. So init to false before each test.

https://github.com/karmada-io/karmada/blob/c8ca014081638fcc8a5eb1e1673c44fbf81bb06a/pkg/search/proxy/controller_test.go#L328-L348

Reply:
Oh, I see.